### PR TITLE
add external_udf_uris support to query()

### PR DESCRIPTION
**Pull request by robin900**
...with docstring and test coverage. Change write_to_table code to use the same
list comprehension to construct userDefinedFunctionResources.

### CODE REVIEW

File: bigquery/client.py [hunk 1]
```diff
@@ -276,7 +276,7 @@ def _insert_job(self, body_object):
             body=body_object
         ).execute()
 
-    def query(self, query, max_results=None, timeout=0, dry_run=False, use_legacy_sql=None):
+    def query(self, query, max_results=None, timeout=0, dry_run=False, use_legacy_sql=None, external_udf_uris=[]):
```
Comment (tylertreat) [inline bigquery/client.py]: We shouldn't use lists as a default value for kwargs (see http://reinout.vanrees.org/weblog/2012/04/18/default-parameters.html). This should default to `None` and check if it's `None` and assign it to an empty list.

**Comment by tylertreat** [comment]
lgtm, just one thing regarding the kwarg. Thanks!

File: bigquery/client.py (hunk 1 above)
Comment (tylertreat) [inline bigquery/client.py]: We shouldn't use lists as a default value for kwargs (see http://reinout.vanrees.org/weblog/2012/04/18/default-parameters.html). This should default to `None` and check if it's `None` and assign it to an empty list.

Comment (robin900) [inline bigquery/client.py]: So I've fixed the existing write_to_table's default arg as well, in a force update.

**Comment by robin900** [comment]
Are you up to making a quick 1.9.1 release to PyPI with this change? I'm using BigQuery-Python in my production code, and would like to avoid sourcing this package from my own git repo.

**Comment by tylertreat** [comment]
@robin900 I will try to get a release made later today.

**Comment by robin900** [comment]
@tylertreat So the bad news is that `query()` -- that is, the synchronous query endpoint in the v2 REST API (POST to /queries) -- doesn't support external UDFs, and there are no plans to add it. Instead, only asynchronous queries -- a POST to the /jobs endpoint -- allow support for UDFs.

I see two possible solutions:
1. Implement `BigQueryClient.query` so that it uses `self._insert_job(query_job_request)`, and does a `wait_on_job` and then returning when job is complete.
2. Doing #1 but only in cases where external_udf_uris are present.

What are your thoughts?

An additional need: external UDFs can be inlined via `inlineCode` instead of `resourceUri`. I have a local change that accepts both `external_udf_uris` and `external_udf_inlines` as arguments to `query()`. Shall I include that additional argument for all relevant client methods?

**Pull request merged**